addons/website_sale_digital/controllers/main.py

```diff
@@ -20,7 +20,7 @@ class WebsiteSaleDigitalConfirmation(WebsiteSale):
     def payment_confirmation(self, **post):
         response = super(WebsiteSaleDigitalConfirmation, self).payment_confirmation(**post)
         order_lines = response.qcontext['order'].order_line
-        digital_content = any(x.product_id.type == 'digital' for x in order_lines)
+        digital_content = any(x.product_id.attachment_count for x in order_lines)
```

Comment:
@JKE-be I saw you removed the review request for `odoo/rd-website`, but this change is worth your attention I think.
I'm not sure it is the correct fix, but I'm sure `x.product_id.type == 'digital'` will always evaluate to `False`. So either this part of the code is dead, or it is bugged.